File: services/spring-api/src/main/kotlin/org/openapitools/model/RecipeIngredient.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.DecimalMax
import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size
import jakarta.validation.Valid
import io.swagger.v3.oas.annotations.media.Schema

/**
 * A measured ingredient has both quantity and unit (\"200 g flour\"); a counted one has a quantity only (\"2 eggs\"); one added to taste has neither (\"salt\"). A unit without a quantity is meaningless and is rejected by the recipe editor, though the contract tolerates it.
 * @param name
 * @param quantity Amount of the ingredient. Omitted when the ingredient is added to taste.
 * @param unit Unit of measurement (e.g. g, ml, cup, tbsp). Omitted when the ingredient is counted as whole items rather than measured.
 */
data class RecipeIngredient(

    @get:Size(min=1)
    @Schema(example = "null", required = true, description = "")
    @get:JsonProperty("name", required = true) val name: kotlin.String,

    @get:DecimalMin(value="0")
    @Schema(example = "null", description = "Amount of the ingredient. Omitted when the ingredient is added to taste.")
    @get:JsonProperty("quantity") val quantity: kotlin.Double? = null,

    @get:Size(min=1)
    @Schema(example = "null", description = "Unit of measurement (e.g. g, ml, cup, tbsp). Omitted when the ingredient is counted as whole items rather than measured.")
    @get:JsonProperty("unit") val unit: kotlin.String? = null
) {

}

File: services/py-help-service/client/cooking_assistant_gen_ai_services_api_internal_client/models/recipe_ingredient.py
from __future__ import annotations

from collections.abc import Mapping
from typing import Any, TypeVar

from attrs import define as _attrs_define

from ..types import UNSET, Unset

T = TypeVar("T", bound="RecipeIngredient")


@_attrs_define
class RecipeIngredient:
    """A measured ingredient has both quantity and unit ("200 g flour"); a counted one has a quantity only ("2 eggs"); one
    added to taste has neither ("salt"). A unit without a quantity is meaningless and is rejected by the recipe editor,
    though the contract tolerates it.

        Attributes:
            name (str):
            quantity (float | Unset): Amount of the ingredient. Omitted when the ingredient is added to taste.
            unit (str | Unset): Unit of measurement (e.g. g, ml, cup, tbsp). Omitted when the ingredient is counted as whole
                items rather than measured.
    """

    name: str
    quantity: float | Unset = UNSET
    unit: str | Unset = UNSET

    def to_dict(self) -> dict[str, Any]:
        name = self.name

        quantity = self.quantity

        unit = self.unit

        field_dict: dict[str, Any] = {}

        field_dict.update(
            {
                "name": name,
            }
        )
        if quantity is not UNSET:
            field_dict["quantity"] = quantity
        if unit is not UNSET:
            field_dict["unit"] = unit

        return field_dict

    @classmethod
    def from_dict(cls: type[T], src_dict: Mapping[str, Any]) -> T:
        d = dict(src_dict)
        name = d.pop("name")

        quantity = d.pop("quantity", UNSET)

        unit = d.pop("unit", UNSET)

        recipe_ingredient = cls(
            name=name,
            quantity=quantity,
            unit=unit,
        )

        return recipe_ingredient
